server/server_test.go

```diff
@@ -170,6 +170,11 @@ func TestInitEngines(t *testing.T) {
 	}
 }
 
+func TestSelfBootstrap(t *testing.T) {
```

Comment:
Might want to add a comment here explaining that the test verifies operation when no bootstrap hosts have been specified.